### PR TITLE
[feature] #3974: Add subcommands into iroha_client_cli to edit domain metadata

### LINKED ISSUE #3974: [suggestion]: Extend functionality of `iroha_client_cli`
### Feature request

After merge of #3963 there is couple of features missing from `iroha_client_cli`.

- [x] Transferring `Domain` between users
- [x] Setting/Removing metadata for `Domain`
- [ ] Minting/Burning trigger repetitions

### Motivation

Help user to express what they what using `iroha_client_cli`.

### Who can help?

@Erigara

### PR DESCRIPTION
**Pull request by VAmuzing**
## Description

Fork of #4054 with a small update to from_str method

### Linked issue

Closes #3974  

### Benefits

There won't be any confusions about why the same method for different structs implemented differently

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: client_cli/src/main.rs [hunk 1]
```diff
@@ -458,6 +461,87 @@ mod domain {
...
     }
+
+    mod metadata {
+        use iroha_client::data_model::{self, domain::DomainId};
+
+        use super::*;
+
+        /// A value wrapper that can be parsed from CLI arguments
+        #[derive(Debug, Clone)]
+        pub struct MetadataValue(data_model::Value);
```
Comment (Erigara) [inline client_cli/src/main.rs:472]: There is [overlap](https://github.com/hyperledger/iroha/pull/4163/files#diff-d2de845c1f7b57cd77a77f1edd12c5f40cecfb0b9aa239cd629e07c1abe965f9R60) between 2 PRs, only one structure should stay.

File: client_cli/src/main.rs [hunk 2]
```diff
@@ -458,6 +461,87 @@ mod domain {
...
+            fn from_str(s: &str) -> Result<Self> {
+                let deser_err_msg = format!("Failed to deserialize `{s}` into value.");
+                let metadata: Value = json5::from_str(s).wrap_err(deser_err_msg)?;
+                Ok(Self(metadata))
+            }
+        }
+
+        /// Edit domain subcommands
+        #[derive(Debug, Clone, clap::Subcommand)]
+        pub enum Args {
```
Comment (Erigara) [inline client_cli/src/main.rs:486]: Getting metadata would also be convinient

Comment (0x009922) [reply client_cli/src/main.rs:486]: I think metadata is shown when you use domain READ operations.

File: client_cli/src/main.rs [hunk 3]
```diff
@@ -458,6 +461,87 @@ mod domain {
...
+        }
+
+        /// Remove metadata into domain by key
+        #[derive(Debug, Clone, clap::Args)]
+        pub struct Remove {
+            /// A domain id from which metadata is to be removed
+            #[arg(short, long)]
+            id: DomainId,
+            #[arg(short, long)]
+            key: Name,
```
Comment (0x009922) [inline client_cli/src/main.rs:533]: Missing docs

File: client_cli/src/main.rs (hunk 1 above)
Comment (0x009922) [reply client_cli/src/main.rs:472]: And I would say that `Value` wrapper from the other PR is more feature-rich in terms of parsing.

Comment (VAmuzing) [reply client_cli/src/main.rs:472]: Then I'll wait for @Asem-Abdelhady PR to be approved & merged and then integrate his changes into this one.

**Review by Erigara — DISMISSED** [review]

File: client_cli/src/main.rs [hunk 4]
```diff
@@ -476,6 +479,74 @@ mod domain {
...
+
+        use super::*;
+
+        /// Edit domain subcommands
+        #[derive(Debug, Clone, clap::Subcommand)]
+        pub enum Args {
+            /// Set metadata into domain
+            Set(Set),
+            /// Remove metadata into domain
+            Remove(Remove),
```
Comment (0x009922) [inline client_cli/src/main.rs:494]:
```suggestion
            /// Set domain metadata
            Set(Set),
            /// Remove domain metadata
            Remove(Remove),
```

**Review by 0x009922 — APPROVED** [review]

**Review by 0x009922 — APPROVED** [review]

**Review by Erigara — APPROVED** [review]

**Pull request merged**